packages/core/contracts-ovm/insured-bridge/implementation/OVM_CrossDomainEnabled.sol

```diff
@@ -0,0 +1,84 @@
+// Copied directly from https://github.com/ethereum-optimism/optimism/blob/7ffb83509c589bd35c2e5f9cb2c6ccbd5f346175/packages/contracts/contracts/optimistic-ethereum/libraries/bridge/OVM_CrossDomainEnabled.sol
+// with a change to the solidity version. Old line is commented out for transparency.
+
+// SPDX-License-Identifier: MIT
+// pragma solidity >0.5.0 <0.8.0;
+pragma solidity ^0.8.0;
+
+/* Interface Imports */
+import "./iOVM_CrossDomainMessenger.sol";
+
+/**
+ * @title OVM_CrossDomainEnabled
+ * @dev Helper contract for contracts performing cross-domain communications
+ *
+ * Compiler used: defined by inheriting contract
+ * Runtime target: defined by inheriting contract
+ */
+contract OVM_CrossDomainEnabled {
+    /*************
+     * Variables *
+     *************/
+
+    // Messenger contract used to send and recieve messages from the other domain.
+    address public messenger;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /**
+     * @param _messenger Address of the CrossDomainMessenger on the current layer.
+     */
+    constructor(address _messenger) {
+        messenger = _messenger;
```

Comment:
Is it guaranteed that this will never change? put differently, is this a special address on the OVM or is it just a normal contract that might be upgraded to a new address later?

Reply:
I think this can change after a "regenesis" event but it should mostly be hardcoded